coi/src/context.rs

```diff
@@ -138,29 +135,33 @@ mod tests {
 
     #[test]
     fn test_compute_score_for_embedding() {
-        let epoch = SystemTime::UNIX_EPOCH;
-        let now = epoch + Duration::from_secs_f32(2. * SECONDS_PER_DAY_F32);
+        let now = Utc::now();
         let mut positive = create_pos_cois([[62., 55., 11.], [76., 30., 80.]]);
-        positive[0].stats.last_view -= Duration::from_secs_f32(0.5 * SECONDS_PER_DAY_F32);
-        positive[1].stats.last_view -= Duration::from_secs_f32(1.5 * SECONDS_PER_DAY_F32);
-
+        positive[0].stats.last_view = now
+            - chrono::Duration::from_std(Duration::from_secs_f32(0.5 * SECONDS_PER_DAY_F32))
+                .unwrap();
+        positive[1].stats.last_view = now
+            - chrono::Duration::from_std(Duration::from_secs_f32(1.5 * SECONDS_PER_DAY_F32))
+                .unwrap();
         let mut negative = create_neg_cois([[6., 61., 6.]]);
-        negative[0].last_view = epoch;
+        negative[0].last_view =
+            now - chrono::Duration::from_std(Duration::from_secs_f32(SECONDS_PER_DAY_F32)).unwrap();
         let cois = UserInterests { positive, negative };
 
+        let embedding = [1., 4., 4.].try_into().unwrap();
         let horizon = Duration::from_secs_f32(2. * SECONDS_PER_DAY_F32);
-
         let score = cois
-            .compute_score_for_embedding(&[1., 4., 4.].try_into().unwrap(), horizon, now)
+            .compute_score_for_embedding(&embedding, horizon, now)
```

Comment:
:pushpin:  Given that we use use `chrono` now pretty much everywhere for `DateTime` we probably should also use it for `Duration`. With the drawback that `chrono` duration support negative duration..

Reply:
in the decay computation we currently rely on non-negative durations as we `unwrap_or_default()` in that case. we could replace it but need an extra check for negative values then.

Reply:
if we want to change this, i guess it's better to do this in a follow up pr. wdyt?

Reply:
yes I think so too

Reply:
i added [ET-4022] to the backlog